### DISCUSSION TITLE
behavior in script setup

### DISCUSSION THREAD

**Discussion by visionm45** [vuejs/vetur#3139] [Ideas]
Hi. I noticed that Vetur does not behave as expected when using a script setup tag.  

The first thing that I noticed is that as this form of creating components does not require anything to be explicitly exported this causes an error to be thrown, complaining about no export, when importing a component in this manner.

The other thing that I noticed is caused by a similar situation.  Since variables, functions and imports made within script setup are exposed to the template without the need to export them then, when using them in the template, Vetur does not recognize that the variables, functions and imports have been used saying that they they have been declared but never read.

Obviously these are not huge problems but I just thought i'd say something since script setup seems to me to be the best way to create and consume components.

**Comment by yoyo930021** [comment]
Vetur doesn't support `<script setup>`.
Please follow #2296.
